### PR TITLE
Bug/gh724 update parameters values grid

### PR DESCRIPTION
**Pull request by omaratstariongroup**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-IME-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-IME [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-IME-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

- Change parameter grid in the grapher into a tree
image

### CODE REVIEW

File: CDP4Composition/ViewModels/CompoundParameterTypeRowViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+        /// <summary>
+        /// Gets or sets the <see cref="SortedList{TKey,TValue}" /> of parameters
+        /// </summary>
+        public SortedList<long, ParameterTypeComponent> ParametersList { get; private set; }
+
+        /// <summary>
+        /// Generate a list of <see cref="ParameterRowControlViewModel" /> that represents <see cref="CompoundParameterType" />
+        /// </summary>
+        /// <returns></returns>
```
Comment (lxatrhea) [inline CDP4Composition/ViewModels/CompoundParameterTypeRowViewModel.cs:65]: missing docs

File: CDP4Composition/ViewModels/CompoundParameterTypeRowViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        }
+
+        /// <summary>
+        /// Generates a row of <see cref="ParameterRowControlViewModel" />
+        /// </summary>
+        /// <param name="parameter">
+        /// The <see cref="ParameterTypeComponent" /> used to generate a
+        /// <see cref="CompoundParameterType" /> row
+        /// </param>
+        /// <returns></returns>
```
Comment (lxatrhea) [inline CDP4Composition/ViewModels/CompoundParameterTypeRowViewModel.cs:86]: missing docs

File: CDP4Composition/ViewModels/ParameterRowControlViewModel.cs [hunk 3]
```diff
@@ -160,13 +178,23 @@ public string RowType
         /// </summary>
         public Option ActualOption { get; private set; }
 
+        /// <summary>
+        /// Gets or sets a list of <see cref="ParameterRowControlViewModel"/>
+        /// </summary>
+        public ReactiveList<ParameterRowControlViewModel> ContainedRows { get; set; }
+
+        public ParameterRowControlViewModel()
```
Comment (lxatrhea) [inline CDP4Composition/ViewModels/ParameterRowControlViewModel.cs:186]: missing docs

File: CDP4Composition/ViewModels/StateDependenceRowViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            };
+
+            return row;
+        }
+
+        /// <summary>
+        /// Gets the matching <see cref="IValueSet"/> for a given <see cref="ActualFiniteState"/> Iid
+        /// </summary>
+        /// <param name="iid">The <see cref="Guid"/> for the current <see cref="ActualFiniteState"/></param>
+        /// <returns></returns>
```
Comment (lxatrhea) [inline CDP4Composition/ViewModels/StateDependenceRowViewModel.cs:91]: missing docs

File: CDP4Composition/ViewModels/ParameterRowControlViewModel.cs [hunk 5]
```diff
@@ -181,18 +209,65 @@ private void UpdateProperties()
...
             this.RowType = this.Parameter.ClassKind.ToString();
+
+            this.UpdateContainedRows();
+        }
+
+        /// <summary>
+        /// Converts the <see cref="ValueArray{T}"/> into a string to be displayed
+        /// </summary>
+        /// <param name="valueArray"></param>
+        /// <returns></returns>
```
Comment (lxatrhea) [inline CDP4Composition/ViewModels/ParameterRowControlViewModel.cs:235]: missing docs

**Review by lxatrhea — APPROVED** [review]

**Review by samatrhea — CHANGES_REQUESTED** [review]

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**